js2py/base.py

```diff
@@ -608,6 +609,16 @@ def to_string(self):
             elif self.is_infinity():
                 sign = '-' if self.value < 0 else ''
                 return Js(sign + 'Infinity')
+            elif not (-7 < floor(log10(abs(self.value))) < 21):
+                # Remove the sign from negative zero
+                x = 0 if self.value == 0 else self.value
```

Comment:
The zero check can be removed from here. It's leftovers from `Number.prototype.toExponential`.